src/platform/user/profile/reducers/index.js

```diff
@@ -56,8 +57,10 @@ const initialState = {
 
 function profileInformation(state = initialState, action) {
   switch (action.type) {
-    case UPDATE_PROFILE_FIELDS:
-      return merge(state, action.newState);
+    case UPDATE_PROFILE_FIELDS: {
+      const newState = mapRawUserDataToState(action.payload);
+      return merge(state, newState);
+    }
 
```

Comment:
I moved the actual mapping into the reducer for consistency across vets-website.